src/compat/posix/time/strptime.c

```diff
@@ -0,0 +1,322 @@
+			}
+		}
+		if (month == 12) {
+			return 0;
+		}
+		tm->tm_mon = month;
+		return count;
+	}
+	return 0;
+}
+
+static int def_format(const char *format, const char *s, struct tm *tm) {
+
+	int n_leng = num_length(s);
+	int w_leng = word_length(s);
+	int date, res, res_equi;
+	int i = 0;
+
+	switch (*format) {
+		case 'm':
+		case 'w':
+		case 'y':
+		case 'Y':
+		case 'H':
+		case 'I':
+		case 'M':
+		case 'S':
+		case 'j':
+		case 'd':
+		case 'e':
+		case 'C':
+		case 'U':
+		case 'W':
+			if (n_leng == 0) {
+				return 0;
+			}
+			date = atoi(s);
+			break;
+	}
+
+	switch (*format) {
+		case 'a':
+		case 'A':
+		case 'b':
+		case 'B':
+		case 'h':
+			if ((w_leng < 3) || (w_leng > 9)) {
+				return 0;
+			}
+			res = cmp_symb(s, format, tm, w_leng);
+			if (res == 0) {
+				return 0;
+			}
+			return res;
+
+		case 'm':
+			if ((date > 12) || (date == 0)) {
+				return 0;
+			}
+			tm->tm_mon = date - 1;
+			return n_leng;
+
+		case 'w':
+			if ((date > 6)) {
+				return 0;
+			}
+			tm->tm_wday = date;
+			return n_leng;
+
+		case 'y':
+			if ((date > 99)) {
+				return 0;
+			}
+			if ((date > 68) && (date < 100)) {
+				tm->tm_year = date;
+			} else {
+				tm->tm_year = date + 100;
+			}
+			return n_leng;
+
+		case'Y':
+			if (n_leng > 4) {
+				return 0;
+			}
+			tm->tm_year = date - 1900;
+			return n_leng;
+
+		case'H':
+			if ((date > 23)) {
+				return 0;
+			}
+			tm->tm_hour = date;
+			return n_leng;
+
+		case'I':
+			if ((date > 12) || (date == 0)) {
+				return 0;
+			}
+			tm->tm_hour = date;
+			return n_leng;
+
+		case 'M':
+			if ((date > 59)) {
+				return 0;
+			}
+			tm->tm_min = date;
+			return n_leng;
+
+		case 'S':
+			if ((date > 60)) {
+				return 0;
+			}
+			tm->tm_sec = date;
+			return n_leng;
+
+		case 'j':
+			if ((date > 366) || (date == 0)) {
+				return 0;
+			}
+			tm->tm_yday = date - 1;
+			return n_leng;
+
+		case 'd':
+		case 'e':
+			if ((date > 31) || (date == 0)) {
+				return 0;
+			}
+			tm->tm_mday = date;
+			return n_leng;
+
+		case 'C':
+			if ((date > 99)) {
+				return 0;
+			}
+			tm->tm_year = date * 100 - 1900;
+			return n_leng;
+
+		case 'U':
+		case 'W':
+			if ((date > 53)) {
+				return 0;
+			}
+			return n_leng;
+
+		case 'n':
+		case 't':
+			if (s[0] == ' ') {
+				return 1;
+			}
+			return 0;
+
+		case 'R':
+		case 'T':
+			res_equi = def_format("H", &s[0], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			if (s[res_equi] != ':' ) {
+				return 0;
+			}
+			res_equi = 0;
+			res_equi = def_format("M", &s[i + 1], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			if (format[0] == 'R') {
+				return res_equi;
+			}
+			if (s[res_equi] != ':' ) {
+				return 0;
+			}
+			res_equi = 0;
+			res_equi = def_format("S", &s[i + 1], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			return res_equi;
+
+		case 'D':
+			res_equi = def_format("m", &s[0], tm);
+			if (res_equi == 0) {
+				return 0;
+			}
+			if (s[res_equi] != '/' ) {
+				return 0;
+			}
+			res_equi = 0;
```

Comment:
Excessive statement (`res_equi` is overwritten on the next line).

Reply:
I deleted excessive statements.
Thank you!